platform/engineering/request-for-comment/2022-05-05-mobile-route-naming-conventions.md

```diff
@@ -0,0 +1,201 @@
+
+## Motivation
+Keeping naming consistent will make using the Mobile API much easier to use and is generally a good practice for maintainability. 
+
+## Design
+Version all endpoint that do not meet the naming convention as follows:
+```
+namespace :health do
+  get '/appointment-requests/:appointment_request_id/messages', to: 'appointment_request_messages#index'
+  get '/appointments', to: 'appointments#index'
+  put '/appointments/:id/cancel', to: 'appointments#cancel'
+  get '/appointments/community-care/:service_type/eligibility', to: 'community_care_eligibility#show'
+  get '/appointments/va/eligibility', to: 'veterans_affairs_eligibility#show'
+  get '/appointments/facility/eligibility', to: 'facility_eligibility#index'
+  post '/appointment', to: 'appointments#create'
+  get '/community-care-providers', to: 'community_care_providers#index'
+  end
+
+namespace :benefits do
+  get '/appeal/:id', to: 'claims_and_appeals#get_appeal'
+  get '/claims-and-appeals-overview', to: 'claims_and_appeals#index'
+  get '/claim/:id', to: 'claims_and_appeals#get_claim'
+  post '/claim/:id/documents', to: 'claims_and_appeals#upload_document'
+  post '/claim/:id/documents/multi-image', to: 'claims_and_appeals#upload_multi_image_document'
+  post '/claim/:id/request-decision', to: 'claims_and_appeals#request_decision'
+  end
+
+  get '/maintenance-windows', to: 'maintenance_windows#index'
+  get '/push/:endpoint_sid/prefs', to: 'push_notifications#get_prefs'
+  put '/push/:endpoint_sid/prefs', to: 'push_notifications#set_pref'
+
+scope :messaging do
+   scope :health do
+      resources :message_drafts, path: 'message-drafts', only: %i[create update], defaults: { format: :json } do
+        post ':reply_id/reply-draft', on: :collection, action: :create_reply_draft, as: :create_reply
+        put ':reply_id/reply-draft/:draft_id', on: :collection, action: :update_reply_draft, as: :update_reply
+      end
+   end
+end
+```
+Notes: 
+- Only endpoints being changed are shown above
+- All of these will be versioned from v0 to v1
+- Adding parent products (health, benefits) is adding a namespace so the controllers will be nested in a folder as well.
+
+For the sake of ease of reviewing all endpoints, see below for all mobile endpoint that will not be changed.
+```
+get '/disability-rating', to: 'disability_rating#index'
+get '/health/immunizations', to: 'immunizations#index'
+get '/health/locations/:id', to: 'locations#show'
+get '/letters', to: 'letters#index'
+get '/letters/beneficiary', to: 'letters#beneficiary'
+post '/letters/:type/download', to: 'letters#download'
+get '/messaging/health/messages/signature', to: 'messages#signature'
+get '/military-service-history', to: 'military_information#get_service_history'
+get '/payment-history', to: 'payment_history#index'
+get '/payment-information/benefits', to: 'payment_information#index'
+put '/payment-information/benefits', to: 'payment_information#update'
```

Comment:
Maybe make Payment parent product?

Reply:
or put the payment information endpoints under benefits

Reply:
Payment information isn't a descriptive name to begin with 😄 as it's easily confused with payment history (which I just did before editing this comment 😀). But yes, I think it should go under benefits.

Reply:
is payment history exclusively used for benefits as well?